### DISCUSSION TITLE
Help Choosing which format to download

### DISCUSSION THREAD

**Discussion by Logan82461** [aclap-dev/video-downloadhelper#593] [Help and Q&A]
### Your question

Hello, tried a search and didnt see an answer but maybe I am searching wrong. 
if this group is only for error questions and not general video download helper questions I am sorry.
Just curious which file I should download for the best picture quality.
MKV or MP4? tried searching google and some said MKV is a better file but found some that said no difference. 
Then there is instances such as the one I am looking at right now. 
Options are 
MP4 1080P 1804x1080 5MB/s or
MKV 1080P 1804x1080 4MB/s 

I am assuming the MP4 but would like some imput. 
If both were equal, both 4MB/s which should I choose then? the MKV or the MP4.

Thanks for you help.

### Addon & CoApp version and details

info not needed for question

**Comment by paulrouget** [comment]
5Mbs has more quality than 4Mbs. MP4 or MKV changes nothing.

I'd go for: MP4 1080P 1804x1080 5MB/s or

**Comment by GrampaWildWilly** [comment]
My observation has been that MP4 & MKV report bit rates differently.  I believe you can't compare, for example, an MP4 at resolution 1920x1080 at 5Kbps against an MKV at resolution 1920x1080 at 4Kbps.  The 5K & the 4K just mean something different because of the different format of the content.  The 2 files would probably look exactly the same when you play them back.

In general, MP4 is neither better nor worse "quality" than MKV.  I put quality in quotation marks because people so often confuse quality & resolution, a problem that is just aggravated by YouTube using the word "quality" where they should be using "resolution."  At a given resolution, they will look the same.  Of course, higher resolutions look better than lower resolutions.

Now suppose you have an MP4 at 1920x1080 at 4Kbps & another MP4 at 1920x1080 at 5Kbps.  Those you can compare.  The one with the higher bit rate should be a higher quality.  Here, the word "quality" applies.  You're comparing 2 MP4s at the same resolution but different bit rates.  The bit rate indicates the quality of the image.  The difference is probably barely visible, but the higher bit rate would be of a higher quality.  The differences become more visible as the bit rates diverge, like 1Kbps vs 7Kbps, other factors like the file type & resolution being the same.

Now suppose you have an MP4 at 1280x720 & an MKV at 1920x1080.  The MKV will probably look better because it's of a higher resolution.  Not because MKV is inherently better than MP4, but because of the different resolutions.  But if you dig deeper & discover that the MP4 has a bit rate around 25Kbps (which would be extraordinarily high) & the MKV has a bit rate around 1Kbps (which would be extraordinarily low), you might be able to see that despite the lower resolution, the image you get from the MP4 is actually of a higher quality.

This is a slippery topic.  There are no pat answers.

**Reply by GrampaWildWilly** [reply]
There's another issue.  VDH reports bit rates in M.  When you look at file properties in Windows & in ffprobe, they report bit rates in K.  In my post above, I uniformly quote bit rates in K because of the Windows/ffprobe behavior.  Those are the numbers I am most used to looking at.  The numbers reported by VDH are kind of a bit baffling to me.  These seem to be numbers reported in HLS master manifests on parameters like BANDWIDTH.  I have not figured out how to correlate the BANDWIDTH numbers in an HLS master manifest with the bit rates reported on the file in Windows after you download it.  The numbers are in different units, M vs K.

Like this wasn't already confusing enough.

**Reply by paulrouget** [reply]
Some science ahead! 😆 

Videos are compressed with **codecs**. Most common codecs are H264 and VP9. **They are all good.**

MP4, MKV and WebM are file **containers**, not codecs. It's just the way the codec-compressed video is arranged inside the file. But the quality or size of the compression is unrelated to the container. Packaging is just not important anymore nowadays. It literally changes nothing. **They are all good.**

The resolution (1280x720, 1920x1080, 720p, 1080p, …) is the size of the video in pixels. `1080p` means "the video height is 1080 pixels). It's just that, the size in pixels, and doesn't guaranty a good quality. You can have a crisp 1280x720 and a sluggish 1920x1080. All screens nowadays are bigger than 720p. So it's recommended to go for at least 1080p at least.

The bitrate ("5MB/s") is actually the most important piece, and it's an exact measurement of the quality. It's basically how much "decoded pixels end up on your screen per seconds". So, literally, **quality**.

**Reply by GrampaWildWilly** [reply]
Here's an example.  I downloaded something this afternoon.  The ffprobe on it reported a variant_bitrate of 4056800.  This would be the number VDH reports.  In this case, VDH would probably report it as 4M, possibly 4.1M.  After I downloaded it, Windows file properties reported the bit rate as 3497Kbps.  That's 4 million vs 35 hundred thousand, more or less.  It's like ffprobe & Windows are speaking different languages.  I pay more attention to the file properties of files on my system.

**Reply by GrampaWildWilly** [reply]
Aha.  So the bit rate is kind of a leveler between MP4 & MKV.  Regardless of container, the higher bit rate reported by VDH is the one you want, given the same resolution.

That still leaves me confused between those bit rates reported by VDH before it downloads & the bit rates in the file properties after I've got the file.

**Reply by Logan82461** [reply]
thanks for all the replies. I totally get resolution sizes and have been going with the highest resolution and, if there were matching resolutions but one with a higher bitrate, ie 3 or 5, I have been choosing the higher bit rate. I have recently started looking for old classics from the 30s to 70s to download as well. I really prefer them to all be 1080 because I watch most of them on a large screen via a JVC RS 400 projector. I have been downloading some 720P and I am going to check a couple of them out today to see how they look and if I should continue to do any more. 
Thanks again for all the replies. 
I have now quit paying attention to the container and am just focusing on resolution first, then bit rate second for matching res.
